project/cmake/scripts/common/projectmacros.cmake

```diff
@@ -50,10 +50,32 @@ endfunction()
 # Copied from FindGTest.cmake 
 # Thanks to Daniel Blezek <blezek@gmail.com> for the GTEST_ADD_TESTS code
 function(GTEST_ADD_TESTS executable extra_args)
+    get_property(XBMC_GENERATED_TEST_FILES GLOBAL PROPERTY XBMC_GENERATED_TEST_FILES)
```

Comment:
yes, there will be an impact. consequence of getting things wrong in the first place, but let's fix it properly.
the first is what i had in mind. whoever creates the CMakeListst.txt for the test library simply needs to be aware.